src/VisualStudio/Core/Def/Implementation/Venus/SimpleContainedDocument.cs

```diff
@@ -0,0 +1,161 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.Experiment;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.Venus
+{
+    using Workspace = Microsoft.CodeAnalysis.Workspace;
+
+    /// <summary>
+    /// An IVisualStudioDocument which represents the secondary buffer to the workspace API.
+    /// </summary>
+    internal sealed class SimpleContainedDocument : ForegroundThreadAffinitizedObject, IVisualStudioHostDocument
+    {
```

Comment:
this is also exact dup of StandardTextDocument - code around open/closing the file in editor. the way razor/venus works in open file is different than regular text file and we already have a way to deal with it. so I am not going to touch that part here. but made this only for closed file case.

*NOTE*, file must be removed from us before file is open.

Reply:
for preview, we will create buffer and view ourselves, not related to invisible editor. we only use that to modify content which it currently doesn't support.